dash_auth/auth.py

```diff
@@ -1,27 +1,40 @@
 from __future__ import absolute_import
 from abc import ABC, abstractmethod
-from typing import Optional
+from typing import Optional, Union
 
 from dash import Dash
 from flask import request
 
 from .public_routes import (
-    add_public_routes, get_public_callbacks, get_public_routes
+    add_public_routes,
+    get_public_callbacks,
+    get_public_routes,
 )
+from .group_protection import protect_layouts
 
 
 class Auth(ABC):
     def __init__(
         self,
         app: Dash,
         public_routes: Optional[list] = None,
-        **obsolete
+        auth_protect_layouts: Optional[Union[dict, bool]] = False,
+        auth_protect_layouts_kwargs: Optional[dict] = None,
+        page_container: Optional[str] = None,
```

Comment:
You are referencing the `page_container`?

My thinking was you are checking it against every trigger regardless of if this is rendering a layout. This could potentially keep callbacks from rendering, if the pathname is not in a public_routes and the user in not authenticated and the callback is a public callback.

I'm also not sure, but hitting it up against the path needlessly in the event of arbitrary callbacks not designed to go against the layout could slow things down?

Reply:
Yes talking about the `page_container`. I don't think speed would be a consideration there, MapAdapter.test is very fast (micro seconds).
There might be other callbacks that trigger on url change than the ones for pages as well, and those should likely be protected as well.

![image](https://github.com/plotly/dash-auth/assets/20662281/ca8c90eb-57d9-4122-96e6-7f29a7e758b1)